### PR TITLE
Add JNI for `cudf::drop_duplicates`

### LINKED ISSUE #9115: [FEA] Java bindings for drop_duplicates
**Is your feature request related to a problem? Please describe.**
As part of implementing batched full joins in Spark, we need the ability to concatenate join gather maps together and remove any duplicate entries.  libcudf has a `drop_duplicates` function but this has not been exposed in the Java APIs.

**Describe the solution you'd like**
The cudf Java bindings should add a `dropDuplicates` method to `Table` with the corresponding options for what to keep, null ordering, and null equality.

### PR DESCRIPTION
**Pull request by ttnghia**
This adds Java binding for `cudf::drop_duplicates`.

Note that when choosing which duplicate element to keep, only `KEEP_FIRST` or `KEEP_LAST` option can be selected. In other words, this does not support `KEEP_NONE` to remove all duplicate elements.

Closes #9115.

### CODE REVIEW

File: java/src/main/java/ai/rapids/cudf/Table.java [hunk 1]
```diff
@@ -1820,6 +1824,25 @@ public Table filter(ColumnView mask) {
     return new Table(filter(nativeHandle, mask.getNativeView()));
   }
 
+  /**
+   * Copy rows of the current table to an output table if the corresponding rows in the keys columns
+   * are unique. The keys columns are a subset of the current table columns and their indices are
```
Comment (jlowe) [inline java/src/main/java/ai/rapids/cudf/Table.java:1829]: This comment implies only unique values are copied and duplicates are not.

File: java/src/main/java/ai/rapids/cudf/Table.java [hunk 2]
```diff
@@ -1820,6 +1824,25 @@ public Table filter(ColumnView mask) {
...
+   * specified by an input array.
+   *
+   * @param keyColumns Array of indices representing key columns from the current table.
+   * @param keepFirst If it is true, the first element in a sequence of duplicate elements will be
+   *                  copied. Otherwise, copy the last element.
+   * @param nullsEqual Flag to denote whether nulls are treated as equal.
+   * @param nullsBefore Flag to specify whether nulls should appear before or after non-null elements.
```
Comment (jlowe) [inline java/src/main/java/ai/rapids/cudf/Table.java:1836]: This implies there's an ordering that can be relied upon in the resulting table.  Can we elaborate in the description above what the caller can assume about the order of keys in the resulting table?  As the docs are written now, I could see a caller assuming the relative positions of preserved keys are unchanged.

Comment (ttnghia) [reply java/src/main/java/ai/rapids/cudf/Table.java]: Thanks. I have updated the comments, clarifying things.

File: java/src/main/java/ai/rapids/cudf/Table.java [hunk 3]
```diff
@@ -1820,6 +1824,30 @@ public Table filter(ColumnView mask) {
...
+  /**
+   * Copy rows of the current table to an output table such that duplicate rows in the key columns
+   * are ignored (i.e., only one row from the duplicate ones will be copied). These keys columns are
+   * a subset of the current table columns and their indices are specified by an input array.
+   *
+   * Currently, the output table is sorted by key columns, using stable sort. However, this is not
+   * guaranteed in the future.
+   *
+   * @param keyColumns Array of indices representing key columns from the current table.
+   * @param keepFirst If it is true, the first row in a sequence of duplicate rows will be copied.
```
Comment (jlowe) [inline java/src/main/java/ai/rapids/cudf/Table.java:1836]: The "sequence of duplicate rows" could make someone wonder, "what if there are duplicates that aren't sequential rows in the table?"  This is exposing implementation details into the parameter description.  I'd limit this to referring to the first row with a duplicated key vs. last row with a duplicated key and don't bother with describing sequence of rows.

Comment (ttnghia) [reply java/src/main/java/ai/rapids/cudf/Table.java]: Fixed that. Sorry for my language :smile:

**Review by jlowe — APPROVED** [review]

**Comment by ttnghia** [comment]
@gpucibot merge

**Pull request merged**